packages/base/src/utils.ts

```diff
@@ -1,54 +1,80 @@
-const blake2b = require("blake2b");
-const isEqual = require("lodash.isequal");
-const { xxHash32 } = require("js-xxhash");
-const { bytes, number } = require("@ckb-lumos/codec");
-const { BI } = require("@ckb-lumos/bi");
-const blockchain = require("./blockchain");
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import blake2b, { Blake2b } from "blake2b";
+import isEqual from "lodash.isequal";
+import { xxHash32 } from "js-xxhash";
+import { bytes, number, BytesLike } from "@ckb-lumos/codec";
+import { BI, BIish } from "@ckb-lumos/bi";
+import * as blockchain from "./blockchain";
+import { Script, Input } from "./api";
+import { Hash, HexNumber, HexString } from "./primitive";
+
+type CKBHasherOptions = {
+  outLength?: number;
+};
 
 const { bytify, hexify, bytifyRawString } = bytes;
 class CKBHasher {
-  constructor(options = {}) {
+  hasher: Blake2b;
+  outLength: number;
+
+  constructor(options: CKBHasherOptions = {}) {
     const { outLength = 32 } = options;
     this.outLength = outLength;
     this.hasher = blake2b(
       outLength,
-      null,
-      null,
+      undefined,
```

Comment:
in [Blake2b document](https://github.com/emilbayes/blake2b#var-hash--blake2boutlength-key-salt-personal-noassert--false), it says the parameter is optional, means `undefined` is a valid parameter.